Classes/Issues/GithubClient+Issues.swift

```diff
@@ -183,4 +184,16 @@ extension GithubClient {
         }))
     }
     
+    func deleteComment(owner: String, repo: String, commentID: Int, completion: @escaping (Result<Bool>) -> ()) {
+        request(Request(path: "repos/\(owner)/\(repo)/issues/comments/\(commentID)", method: .delete, completion: { (response, _) in
+            // As per documentation this endpoint returns no content, so all we can validate is that
+            // the status code is "204 No Content".
+            if response.response?.statusCode == 204 {
+                completion(.success(true))
+            } else {
+                completion(.error(response.error))
+            }
+        }))
+    }
```

Comment:
Another option for this would be to do all the deletion inside the section controller and then just return `0` for `numberOfItems`. That'd keep any of it from creeping back into the view controller, as well as make local vs remote comments exactly the same. Thoughts?

Reply:
Happy to make the change, but I do feel this will begin to add further complexity where it may not be super obvious where items are being filtered out/in. At the moment we have one place which defines all the models, and we expect something to show for them - this would then say "there should be a comment here" but then later on "but let's not show it".

Get what I mean?

Reply:
Still want this change made @rnystrom? If so will sort it out in morning 😄

Reply:
Ya, let's move it to a 0 item. I'd rather keep the complexity local to the section controller instead of too many wires. Will be much more modular.